tests/e2e/root_v3_test.go

```diff
@@ -0,0 +1,45 @@
+package e2e_test
+
+import (
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+type RootV3Resource struct {
+	Links struct {
+		Self struct {
+			Href string `json:"href"`
+		} `json:"self"`
+	} `json:"links"`
+}
+
+var _ = Describe("TrailingSlash", func() {
```

Comment:
I would recommend several changes to the structure of the test:
1) change the `Describe` test to something like `RootV3` as this test is supposed to cover the `/v3` endpoint
2) Introduce a variable e.g. `requestPath` that is defaulted to `/v3` in `BeforeEach` and have a single request in `JustBeforeEach` for that path. In the top-level `It` just verify that the request succeeds.
3) Introduce a context e.g. `When("the request path ends with a trailing slack"....` whose `BeforeEach` is resetting the `requestPath` to `/v3/`. Then, when the top-level`JustBeforeEach` is executed it would pick up the value with the trailing slash. In that `When` there should be another `It` that verifies that the request still succeeds.

Following this structure would make the test comply with the standard Ginkgo idiom where there is preferably a single `JustBeforeEach` that exercises the code under test, and then multiple contexts with their `BeforeEach`es that alter the arguments to the invoccation under test.